### PR TITLE
Leading zeros and other bit manipulations

### PR DESCRIPTION
**Pull request by TheIronBorn**
#188

### CODE REVIEW

File: src/api/bit_manip.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                        let elems: &mut [$elem_ty] = unsafe {
+                            slice::from_raw_parts_mut(
+                                BYTES.as_mut_ptr() as *mut $elem_ty,
+                                $id::lanes(),
+                            )
+                        };
+                        $id::from_slice_unaligned(elems)
+                    }
+
+                    #[cfg_attr(not(target_arch = "wasm32"), test)] #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test)]
```
Comment (gnzlbg) [inline src/api/bit_manip.rs]: might want to split these into two lines - rustfmt chokes on these

**Comment by gnzlbg** [comment]
For the following architectures you need to:

* fill in an LLVM bug: ideally you just write LLVM-IR in godbolt for one case that works, say in x86_64, compile it for one of these targets, and show the LLVM error.
* open a bug here referencing the LLVM bug
* add a workaround to codegen for these targets that, instead of using the LLVM intrinsic, uses pure rust code to compute these - these workaround should have a FIXME referencing the bug in these repo


It looks like these are not available in aarch64:

```
LLVM ERROR: Cannot select: 0x7fb76ee85270: v8i8 = cttz 0x7fb76ef05270, src/codegen/bit_manip.rs:176:39
  0x7fb76ef05270: v8i8,ch = load<(dereferenceable load 8 from %ir.5)> 0x7fb76ef23758, FrameIndex:i64<2>, undef:i64, src/codegen/bit_manip.rs:176:31
    0x7fb76ee5b138: i64 = FrameIndex<2>
    0x7fb76ee90958: i64 = undef
In function: _ZN110_$LT$packed_simd..Simd$LT$$u5b$u8$u3b$$u20$_$u5d$$GT$$u20$as$u20$packed_simd..codegen..bit_manip..BitManip$GT$4cttz17h3aae67eb204ee427E
error: Could not compile `packed_simd`.
```

And in s390x:

**Comment by TheIronBorn** [comment]
Turns out aarch64 has instructions for ctlz and ctpop. We should add those to `assert_instr` tests.

**Comment by gnzlbg** [comment]
Please ping me when CI is green :) I try to check on this every now and then, but if you don't hear from me, then just ping me :D

**Comment by TheIronBorn** [comment]
The remaining CI failures seem to be unrelated.

**Comment by gnzlbg** [comment]
Also could you open an issue about optimizing the aarch64 intrinsics here (and maybe include there adding assert_instr tests for those) ?

**Comment by TheIronBorn** [comment]
It seemed like every s390x intrinsic was failing so I just disabled them entirely

**Comment by gnzlbg** [comment]
@TheIronBorn do you want to be a maintainer of `packed_simd` ?

**Pull request merged**